static/app/components/profiling/profilesTable.tsx

```diff
@@ -0,0 +1,279 @@
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import Link from 'sentry/components/links/link';
+import Pagination from 'sentry/components/pagination';
+import PerformanceDuration from 'sentry/components/performanceDuration';
+import {IconCheckmark, IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Trace} from 'sentry/types/profiling/core';
+import {defined} from 'sentry/utils';
+import {Container, NumberContainer} from 'sentry/utils/discover/styles';
+import {getShortEventId} from 'sentry/utils/events';
+import {
+  generateFlamegraphSummaryRoute,
+  generateFunctionsRouteWithQuery,
+} from 'sentry/utils/profiling/routes';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+
+const REQUIRE_PROJECT_COLUMNS = new Set(['id', 'project_id', 'transaction_name']);
+
+interface ProfilesTableProps {
+  error: string | null;
+  isLoading: boolean;
+  traces: Trace[];
+  columnOrder?: TableColumnKey[];
+  hideHeader?: boolean;
+  pageLinks?: string | null;
+}
+
+function ProfilesTable(props: ProfilesTableProps) {
+  const location = useLocation();
+
+  return (
+    <Fragment>
+      {!props.hideHeader && (
+        <TableHeader>
+          <SectionHeading>{t('Recent Profiles')}</SectionHeading>
+          <StyledPagination pageLinks={props.pageLinks} size="xsmall" />
+        </TableHeader>
+      )}
+      <GridEditable
+        isLoading={props.isLoading}
+        error={props.error}
+        data={props.traces}
+        columnOrder={(props.columnOrder ?? DEFAULT_COLUMN_ORDER).map(key => COLUMNS[key])}
+        columnSortBy={[]}
+        grid={{renderBodyCell: renderProfilesTableCell}}
+        location={location}
+      />
+    </Fragment>
+  );
+}
+
+function renderProfilesTableCell(
+  column: TableColumn,
+  dataRow: TableDataRow,
+  rowIndex: number,
+  columnIndex: number
+) {
+  return (
+    <ProfilesTableCell
+      column={column}
+      dataRow={dataRow}
+      rowIndex={rowIndex}
+      columnIndex={columnIndex}
+    />
+  );
+}
+
+interface ProfilesTableCellProps {
+  column: TableColumn;
+  columnIndex: number;
+  dataRow: TableDataRow;
+  rowIndex: number;
+}
+
+function ProfilesTableCell({column, dataRow}: ProfilesTableCellProps) {
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const location = useLocation();
+
+  // Not all columns need the project, so small optimization to avoid
+  // the linear lookup for every cell.
+  const project = REQUIRE_PROJECT_COLUMNS.has(column.key)
+    ? projects.find(proj => proj.id === dataRow.project_id)
+    : undefined;
+
+  const value = dataRow[column.key];
+
+  switch (column.key) {
+    case 'id':
+      if (!defined(project)) {
+        // should never happen but just in case
+        return <Container>{getShortEventId(dataRow.id)}</Container>;
+      }
```

Comment:
If this should never happen, should we just throw here so we see the errors? We could also just return before the whole switch statement by doing something like


```tsx
if(REQUIRE_PROJECT_COLUMNS.has(column.key) && !project){
  return <Placeholder/>
}
```